app/services/EventReportService.scala

```diff
@@ -161,44 +162,66 @@ class EventReportService @Inject()(eventReportConnector: EventReportConnector,
                                              eventType: EventType,
                                              pstr: String,
                                              currentVersion: Int)
                                         (implicit headerCarrier: HeaderCarrier, ec: ExecutionContext, request: RequestHeader):JsObject = {
 
-    def getMemberDetails(userAnswers: JsObject) = {
-      Try(
-        (userAnswers \ ("event" + eventType.toString) \ "members").as[JsArray].value.map(_.as[JsObject])
-      ) match {
-        case Failure(exception) =>
-          logger.warn("Could not get member details", exception)
-          IndexedSeq(JsObject(Seq()))
-        case Success(value) => value
+    def newMembersWithChangeInfo(getMemberDetails: JsObject => scala.collection.IndexedSeq[JsObject]) = {
+      val newMembers = getMemberDetails(newUserAnswers)
+
+      newMembers.zipWithIndex.map { case (newMemberDetail, index) =>
+        val oldMemberDetails = oldUserAnswers.map(getMemberDetails).getOrElse(Seq())
+        val oldMemberDetail = Try(oldMemberDetails(index)).toOption
+        val newMemberChangeInfo = generateMemberChangeInfo(
+          oldMemberDetail,
+          newMemberDetail,
+          currentVersion
+        )
+
+        newMemberDetail +
+          ("amendedVersion", JsString(("00" + newMemberChangeInfo.amendedVersion.toString).takeRight(3))) +
+          ("memberStatus", JsString(newMemberChangeInfo.status.name))
       }
     }
 
     apiProcessingInfo(eventType, pstr) match {
       case Some(APIProcessingInfo(apiType, _, _, _)) =>
         apiType match {
+          case ApiType.Api1827 =>
+            def getMemberDetails(userAnswers: JsObject) =
+              (userAnswers \ "event1" \ "membersOrEmployers").as[JsArray].value.map(_.as[JsObject])
+
+            val event1 = {
+              Try(
+                ((newUserAnswers \ "event1").as[JsObject] - "membersOrEmployers") +
+                ("membersOrEmployers", Json.toJson(newMembersWithChangeInfo(getMemberDetails)))
+              ) match {
+                case Failure(exception) =>
+                  logger.warn("Could not get member details for event 1", exception)
+                  Json.toJson(IndexedSeq(JsObject(Seq())))
+                case Success(value) => value
+              }
+            }
+
+            (newUserAnswers - "event1") + ("event1", event1)
           case ApiType.Api1830 =>
-            val newMembers = getMemberDetails(newUserAnswers)
-
-            val newMembersWithChangeInfo = newMembers.zipWithIndex.map { case (newMemberDetail, index) =>
-              val oldMemberDetails = oldUserAnswers.map(getMemberDetails).getOrElse(Seq())
-              val oldMemberDetail = Try(oldMemberDetails(index)).toOption
-              val newMemberChangeInfo = generateMemberChangeInfo(
-                  oldMemberDetail,
-                  newMemberDetail,
-                  currentVersion
-                )
```

Comment:
As we need to get this merged I'll approve but I think it will be cleaner and more concise if we pull the code for 1827 & 1830 into a private def - most of the code is duplicated, just the node name is different (membersOrEmployers vs members)